jetty-http/src/test/java/org/eclipse/jetty/http/GZIPContentDecoderTest.java

```diff
@@ -351,4 +358,97 @@ public void testBigBlockWithExtraBytes() throws Exception
         assertTrue(buffer.hasRemaining());
         assertEquals(data2, StandardCharsets.UTF_8.decode(buffer).toString());
     }
+
+    // Signed Integer Max
+    final long INT_MAX = Integer.MAX_VALUE;
+
+    // Unsigned Integer Max == 2^32
+    final long UINT_MAX = 0xffffffffL;
+
+    @ParameterizedTest
+    @ValueSource(longs = {INT_MAX, INT_MAX + 1, UINT_MAX, UINT_MAX + 1})
+    public void testLargeGzipStream(long origSize) throws IOException
+    {
+        final int BUFSIZE = (1 * 1024 * 1024); // 1MB
+
+        // Create a buffer to use over and over again to produce the uncompressed input
+        byte[] cbuf = "0123456789ABCDEFGHIJKLMOPQRSTUVWXYZ".getBytes(StandardCharsets.UTF_8);
+        byte[] buf = new byte[BUFSIZE];
+        for (int off = 0; off < buf.length; )
+        {
+            int len = Math.min(cbuf.length, buf.length - off);
+            System.arraycopy(cbuf, 0, buf, off, len);
+            off += len;
+        }
+
+        GZIPContentDecoder decoder = new GZIPContentDecoder(BUFSIZE);
+
+        try (GZIPDecoderOutputStream out = new GZIPDecoderOutputStream(decoder);
+             GZIPOutputStream outputStream = new GZIPOutputStream(out, BUFSIZE))
+        {
+            long writtenBytes = 0L;
```

Comment:
`writtenBytes` is the same as `origSize`, so you can drop it.

Reply:
`writtenBytes` tracks the write (pre-encode) side.
`out.decodedByteCount` tracks the read (post-decode) side.

Reply:
When `writtenBytes` is used, it will have the same value as `origSize` so it's unnecessary.

Reply:
I reworked that section.
See updated code.

Reply:
Removed writtenBytes.